pkg/safeguards/preprocessing_helpers.go

```diff
@@ -0,0 +1,113 @@
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/chartutil"
+	"helm.sh/helm/v3/pkg/engine"
+)
+
+// Given a Helm chart directory or file, renders all templates and writes them to the specified directory
+func RenderHelmChart(isFile bool, mainChartPath, tempDir string) error {
+	if isFile { // Get the directory that the Chart.yaml lives in
+		mainChartPath = filepath.Dir(mainChartPath)
+	}
+
+	mainChart, err := loader.Load(mainChartPath)
+	if err != nil {
+		return fmt.Errorf("failed to load main chart: %s", err)
+	}
+
+	loadedCharts := make(map[string]*chart.Chart) // map of chart path to chart object
+	loadedCharts[mainChartPath] = mainChart
+
+	// Load subcharts and dependencies
+	for _, dep := range mainChart.Metadata.Dependencies {
+		// Resolve the chart path based on the main chart's directory
+		chartPath := filepath.Join(mainChartPath, dep.Repository[len("file://"):])
+		chartPath = filepath.Clean(chartPath)
+
+		subChart, err := loader.Load(chartPath)
+		if err != nil {
+			return fmt.Errorf("failed to load chart: %s", err)
+		}
+		loadedCharts[chartPath] = subChart
+	}
+
+	for chartPath, chart := range loadedCharts {
+		valuesPath := filepath.Join(chartPath, "values.yaml") // Enforce that values.yaml must be at same level as Chart.yaml
+		mergedValues, err := getValues(chart, valuesPath)
+		if err != nil {
+			return fmt.Errorf("failed to load values: %s", err)
+		}
+		e := engine.Engine{Strict: true}
+		renderedFiles, err := e.Render(chart, mergedValues)
+		if err != nil {
+			return fmt.Errorf("failed to render chart: %s", err)
+		}
+
+		if err := os.MkdirAll(tempDir, 0755); err != nil {
+			return fmt.Errorf("failed to create output directory: %s", err)
+		}
+
+		// Write each rendered file to the output directory with the same name as in templates/
+		for filePath, content := range renderedFiles {
+			outputFilePath := filepath.Join(tempDir, filepath.Base(filePath))
+			if err := os.WriteFile(outputFilePath, []byte(content), 0644); err != nil {
+				return fmt.Errorf("failed to write manifest file: %s", err)
+			}
+
+		}
+	}
+
+	return nil
+}
+
+// Returns values from values.yaml and release options specified in values.yaml
+func getValues(chart *chart.Chart, valuesPath string) (chartutil.Values, error) {
+	// Load values file
+	valuesFile, err := os.ReadFile(valuesPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read values file: %s", err)
+	}
+
+	vals := map[string]interface{}{}
+	if err := yaml.Unmarshal(valuesFile, &vals); err != nil {
+		return nil, fmt.Errorf("failed to parse values.yaml: %s", err)
+	}
+
+	mergedValues, err := getReleaseOptions(chart, vals)
+	return mergedValues, err
+}
+
+func getReleaseOptions(chart *chart.Chart, vals map[string]interface{}) (chartutil.Values, error) {
+	// Extract release options from values
+	releaseName, ok := vals["releaseName"].(string) //TODO: What do we want to do if a releaseName and namespace is not specified in the values.yaml?
```

Comment:
Can we discover release names? I think if there are multiple options, then we should error out and make the user specify the release name(we can add an argument for that)

If there are no options, then I think we can assume the release name is the name of the containing directory, and the releaseNamespace can just be "default"

Reply:
Or we could have the release name be the chart name?

Reply:
As discussed offline, we can require the user to pass in the release name and namespace in the cli or read from values.yaml. These could be added in the cli options description. Maybe we can add a third option which sets the release name to the name of the containing directory and the namespace to default. That way the user experience will be more seamless. These fields are only really used in metadata.name, metadata.namespace so they shouldn't cause any issues in validation if we add default values anyways.

Reply:
when the user installs a chart, they can specify both a namespace and release name- which we don't have access to without prompting the user
i haven't seen any constraints that can be violated by changing these values, so we could meet in the middle and prompt for release name with a default of the containing directory name, and the same for namespace.
this would allow most users to accept the default, but for those with specific namespaces they can supply this in the prompts

Reply:
I'll keep that in mind for the next PR where we'll integrate changes with validate.go